model/roles.go

```diff
@@ -979,6 +994,98 @@ func (j *JobReference) WriteConfigs(instanceGroup *InstanceGroup, lightOpinionsP
 	return json.MarshalIndent(config, "", "    ") // 4-space indent
 }
 
+// validateVariableDescriptions tests whether all variables have descriptions
+func validateVariableDescriptions(roleManifest *RoleManifest) validation.ErrorList {
+	allErrs := validation.ErrorList{}
+
+	for _, variable := range roleManifest.Configuration.Variables {
+		if variable.Description == "" {
+			allErrs = append(allErrs, validation.Required(variable.Name,
+				"Description is required"))
+		}
+	}
+
+	return allErrs
+}
+
+// validateSortedTemplates tests that all templates are sorted in alphabetical order
+func validateSortedTemplates(roleManifest *RoleManifest) validation.ErrorList {
+	allErrs := validation.ErrorList{}
+
+	previousKey := ""
+
+	for _, templateDef := range roleManifest.Configuration.Templates {
+		key := templateDef.Key.(string)
+
+		if previousKey != "" && previousKey > key {
+			allErrs = append(allErrs, validation.Forbidden(previousKey,
+				fmt.Sprintf("Template key does not sort before '%s'", key)))
+		}
+
+		previousKey = key
+	}
+
+	return allErrs
+}
+
+// validateScripts tests that all scripts exist and that all referenced scripts exist
+func validateScripts(roleManifest *RoleManifest) validation.ErrorList {
+	allErrs := validation.ErrorList{}
+	scriptDir := filepath.Join(filepath.Dir(roleManifest.manifestFilePath), "scripts")
+	usedScripts := map[string]bool{}
+
+	for _, instanceGroup := range roleManifest.InstanceGroups {
+		for _, script := range instanceGroup.Scripts {
+			fullScriptPath := filepath.Join(scriptDir, script)
+
+			if _, err := os.Stat(fullScriptPath); err != nil {
+				allErrs = append(allErrs, validation.Invalid(
+					fmt.Sprintf("%s script", instanceGroup.Name),
+					script,
+					err.Error()))
+			}
+
+			usedScripts[fullScriptPath] = true
+		}
+
+		for _, script := range instanceGroup.EnvironScripts {
+			fullScriptPath := filepath.Join(scriptDir, script)
+
+			if _, err := os.Stat(fullScriptPath); err != nil {
+				allErrs = append(allErrs, validation.Invalid(
+					fmt.Sprintf("%s env script", instanceGroup.Name),
+					script,
+					err.Error()))
+			}
+
+			usedScripts[fullScriptPath] = true
+		}
+	}
+
+	err := filepath.Walk(scriptDir,
+		func(path string, info os.FileInfo, err error) error {
+			if err != nil {
+				return err
+			}
+
+			if info.IsDir() {
+				return nil
+			}
+
+			if _, ok := usedScripts[path]; !ok {
```

Comment:
This is probably the same as `if !usedScripts[path]`, because all entries are `true`.

Reply:
Not necessarily if `usedScripts` is a map. Then the entry may not exist, and the code as-is treats this non-presence as `false`. Your proposed code may panic instead ?

Reply:
No, the zero value of a boolean is false. https://play.golang.org/p/L1PBhJI9v2o
This would only panic if the map was not allocated.

Reply:
Ok. Retracting my comment